Core/src/Propagator/detail/PointwiseMaterialInteraction.cpp

```diff
@@ -38,9 +38,11 @@ void PointwiseMaterialInteraction::covarianceContributions(
   }
   // TODO just ionisation loss or full energy loss?
   if (energyLoss) {
-    const float sigmaQoverP =
+    // Squaring float sigmaQoverP was observed to lead to underflow.
+    // Square in double precision and then cast back to float.
+    const double sigmaQoverP =
         computeEnergyLossLandauSigmaQOverP(slab, mass, qOverP, absQ);
-    varianceQoverP = sigmaQoverP * sigmaQoverP;
+    varianceQoverP = static_cast<float>(sigmaQoverP * sigmaQoverP);
```

Comment:
The new random numbers seem to lead to a case where this calculation produces a float underflow.

My first attempt is to do the square in double and then cast back to float. An alternative could be to check `sigmaQoverP` and just set the square to zero if it's too small.

Any other ideas @andiwand?

Reply:
my argument would be that the issue was there before but hidden by the seed so it seems fine to me to move back to that situation. to tackle this problem we need to understand why we end up in this situation in the first place I believe?